### PR TITLE
Blog index page cleanup + ajax

### PR DESCRIPTION
**Pull request by boutell**
Fix various issues with the tag filters, use the correct query parameter for multiple tag filtering (tags), don't remove the first tag if enter key pressed in search field, show value in search field, AJAX the blog index page

### CODE REVIEW

File: lib/modules/apostrophe-assets/public/js/modules/pieces-pages-search.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+$(function() {
+  $('body').on('keypress', '[data-apos-ajax-context] input[name="search"]', function(e) {
```
Comment (boutell) [inline lib/modules/apostrophe-assets/public/js/modules/pieces-pages-search.js:2]: This is a delegated event handler so it's safe to do it outside "enhance".

File: lib/modules/apostrophe-assets/public/js/popups/tags.js [hunk 2]
```diff
@@ -53,10 +58,8 @@ $(function () {
     var $clone = $tag.clone();
     $clone.removeAttr('data-tag-picker-selected');
     $clone.on('click', function (e) {
-      var $this = $(this);
       e.preventDefault();
-      $context.find('[data-tag-picker-selected]button[value="' + $this.val() + '"]').attr('data-tag-picker-selected', 'false');
-      $this.remove();
+      removeTag($clone, $context);
```
Comment (boutell) [inline lib/modules/apostrophe-assets/public/js/popups/tags.js:62]: Without this the implementation was incomplete, it was visually removed on click but not removed from the list on submit. Also hey less code and seems to work?

File: lib/modules/apostrophe-assets/public/js/modules/pieces-pages-search.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+    if (e.which === 13) {
+      // Do not fire the first button in the form (default behavior), which would
+      // remove the first selected tag
+      event.preventDefault();
```
Comment (stuartromanek) [inline lib/modules/apostrophe-assets/public/js/modules/pieces-pages-search.js]: `event` is not here, it is `e` ?

**Review by stuartromanek — COMMENTED** [review]
just this one check

**Review by stuartromanek — APPROVED** [review]

**Pull request merged**